src/query/functions/src/scalars/decimal.rs

```diff
@@ -460,36 +618,60 @@ pub fn register(registry: &mut FunctionRegistry) {
     });
 }
 
+macro_rules! single_decimal128_to_float {
```

Comment:
can we use https://github.com/andylokandy/databend/blob/811ff87f748ae5ba325afa6c930605cc40fa1ab8/src/query/expression/src/types/decimal.rs#L200?

Reply:
Yes